click/core.py

```diff
@@ -129,6 +129,27 @@ def sort_key(item):
 
     return sorted(declaration_order, key=sort_key)
 
+class ParameterSource(object):
```

Comment:
nit: Should this be named `OptionSource`? Positional arguments can't be set via the environment, have default values, etc. so this is only relevant for options. I think this was the intention since the docstring says "indicates the source of a command line option" already. https://click.palletsprojects.com/en/7.x/parameters/#differences

Reply:
https://click.palletsprojects.com/en/7.x/arguments/#environment-variables states that arguments can be set via the environment, will change the docs to parameter from option.